src/algebra/algebra/subalgebra.lean

```diff
@@ -166,6 +166,30 @@ instance to_comm_ring {R A}
   [comm_ring R] [comm_ring A] [algebra R A] (S : subalgebra R A) :
   comm_ring S := S.to_subring.to_comm_ring
 
+instance to_ordered_semiring {R A}
+  [comm_semiring R] [ordered_semiring A] [algebra R A] (S : subalgebra R A) :
+  ordered_semiring S := S.to_subsemiring.to_ordered_semiring
+instance to_ordered_comm_semiring {R A}
+  [comm_semiring R] [ordered_comm_semiring A] [algebra R A] (S : subalgebra R A) :
+  ordered_comm_semiring S := subsemiring.to_ordered_comm_semiring S
+instance to_ordered_ring {R A}
+  [comm_ring R] [ordered_ring A] [algebra R A] (S : subalgebra R A) :
+  ordered_ring S := S.to_subring.to_ordered_ring
+instance to_ordered_comm_ring {R A}
+  [comm_ring R] [ordered_comm_ring A] [algebra R A] (S : subalgebra R A) :
+  ordered_comm_ring S := S.to_subring.to_ordered_comm_ring
+
+instance to_linear_ordered_semiring {R A}
+  [comm_semiring R] [linear_ordered_semiring A] [algebra R A] (S : subalgebra R A) :
+  linear_ordered_semiring S := S.to_subsemiring.to_linear_ordered_semiring
+/-! There is no `linear_ordered_comm_semiring`. -/
```

Comment:
```suggestion
/- There is no `linear_ordered_comm_semiring`. -/
```
I don't think this should be a module docstring, right?

Reply:
I think it should, so that when reading the docs the users see
```
foo_instance
bar_instance

there is no `baz_class`

foobar_class
```

Reply:
Ok, fair enough